scipy/integrate/_quadrature.py

```diff
@@ -1029,20 +1031,22 @@ def cumulative_simpson(y, *, x=None, dx=1.0, axis=-1, initial=None):
     y = _ensure_float_array(y)
 
     # validate `axis` and standardize to work along the last axis
+    original_y = y.copy()
     original_shape = y.shape
     try:
         y = np.swapaxes(y, axis, -1)
     except IndexError as e:
         message = f"`axis={axis}` is not valid for `y` with `y.ndim={y.ndim}`."
         raise ValueError(message) from e
+    
+    if y.shape[-1] == 0:
+        raise ValueError("At least 1 point is required along axis.")
```

Comment:
```suggestion
        raise ValueError("At least one point is required along `axis`.")
```
since you're referring to the variable. And while we're at it, a[ common rule of thumb](https://grammarly.com/blog/when-to-spell-out-numbers/) is to spell out numbers less than ten. Same in the documentation updates at the top.

And why not put this error in `cumulative_trapezoid` so both functions raise the desired error?